src/components/phone-input/PhoneInput.tsx

```diff
@@ -62,29 +65,38 @@ export const PhoneInput = ({
       return;
     }
 
-    const errorCode = await onSubmitPromise(phoneValidation.phoneNumber);
-    if (!errorCode) {
+    const result = await onSubmitPromise(phoneValidation.phoneNumber);
+
+    setIsSubmitting(false);
+
+    if (!result) {
```

Comment:
We might simplify the conditions by just using:

```
if (isError(result)) {
        setError(result);
      } else {
        setError(undefined);
        onSubmitAction(phoneValidation.phoneNumber, result);
      }
```

Reply:
thanks!

Reply:
Updated!